scripts/Diffraction/isis_powder/osiris_routines/osiris_algs.py

```diff
@@ -228,23 +239,47 @@ def run_diffraction_focussing(run_number, drange_sets, calfile, van_norm=False,
     return focused
 
 
-def get_run_details(run_number_string, inst_settings, is_vanadium_run):
+def create_drange_sets(run_number_string, inst, file_ext):
+    run_number_list = common.generate_run_numbers(run_number_string=run_number_string)
+
+    drange_sets = {}
+    for run_number in run_number_list:
+        ws_name = inst._generate_input_file_name(run_number=run_number, file_ext=file_ext if file_ext else "")
+        ws = common.load_file(ws_name)
+        drange = get_osiris_d_range(ws)
+
+        if drange not in drange_sets:
+            drange_sets[drange] = DrangeData(drange)
+
+        drange_sets[drange].add_sample(run_number)
+
+    return drange_sets
+
+
+def get_run_details(run_number_string, inst_settings, is_vanadium_run, drange=None):
     all_run_numbers = get_cal_mapping_dict(run_number_string, inst_settings.cal_mapping_path)
-    empty_runs = _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers, key="empty_run_numbers")
-    vanadium_runs = _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers, key="vanadium_run_numbers")
 
-    grouping_file_name = inst_settings.grouping
+    if not drange:
+        empty_can_runs = (
+            _get_run_numbers_for_key(current_mode_run_numbers=all_run_numbers, key="empty_run_numbers")
```

Comment:
There's an unfortunate name clash here because Osiris is currently using the setting called `empty_run_numbers` in its yaml file to refer to empty can runs. Whereas all the other instruments use the same setting in the .yaml file to refer to empty instrument runs. Could you edit the Osiris .yaml and this line to use the name `empty_can_run_numbers` instead? I see you've already made some variable name changes to improve things in this area but this would make it clearer still

Reply:
Renamed